lib/beetle/subscriber.rb

```diff
@@ -168,45 +168,44 @@ def resume(queue_name)
     def create_subscription_callback(queue_name, amqp_queue_name, handler, opts)
       server = @server
       lambda do |header, data|
+        msg_id = header.attributes[:message_id]
         if channel(server).closing?
-          logger.info "Beetle: ignoring message #{header.attributes[:message_id]} (#{header.attributes[:timestamp]}) since channel to server #{server} already closed"
+          timestamp = header.attributes[:timestamp]
+          logger.info "Beetle: ignored message #{msg_id}(#{timestamp}) since channel to server #{server} was already closed"
           return
         end
         begin
-          logger.debug("Beetle: #{header.attributes[:message_id]} (#{header.attributes[:timestamp]}) received")
-          # logger.debug "Beetle: received message"
           processor = Handler.create(handler, opts)
+          processor.pre_process
           message_options = opts.merge(:server => server, :store => @client.deduplication_store)
           m = Message.new(amqp_queue_name, header, data, message_options)
           result = m.process(processor)
           if result.reject?
-            logger.debug("Beetle: #{header.attributes[:message_id]} (#{header.attributes[:timestamp]}) rejected")
             if @client.config.dead_lettering_enabled?
               header.reject(:requeue => false)
             else
               sleep 1
               header.reject(:requeue => true)
             end
+            logger.debug "Beetle: rejected #{msg_id} RC: #{result.name}"
```

Comment:
Shouldn't we have the timestamp here as well (as we are adding it to other logs of the `msg_id`)?

Reply:
My reasoning is we don't need to add the timestamp to every log line. but the code makes sure it's logged once (at the beginning).